api/v1/nodeptpdevice_types.go

```diff
@@ -21,47 +21,73 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+//+kubebuilder:printcolumn:name="HwConfig",type="string",JSONPath=".status.hwconfig.status"
+
+// NodePtpDevice is the Schema for the nodeptpdevices API
+type NodePtpDevice struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NodePtpDeviceSpec   `json:"spec,omitempty"`
+	Status NodePtpDeviceStatus `json:"status,omitempty"`
+}
 
 // NodePtpDeviceSpec defines the desired state of NodePtpDevice
 type NodePtpDeviceSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
 }
 
 type PtpDevice struct {
-	Name    string `json:"name,omitempty"`
+	// Name is the name of the PTP device.
+	// It is a unique identifier for the device.
+	// +required
+	Name string `json:"name,omitempty"`
+
+	// Profile is the PTP profile associated with the device.
+	// This profile defines the PTP configuration settings for the device.
+	// +required
 	Profile string `json:"profile,omitempty"`
 }
 
-// NodePtpDeviceStatus defines the observed state of NodePtpDevice
-type NodePtpDeviceStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-	Devices  []PtpDevice `json:"devices,omitempty"`
-	Hwconfig []HwConfig  `json:"hwconfig,omitempty"`
-}
-
 type HwConfig struct {
-	DeviceID string              `json:"deviceID,omitempty"`
-	VendorID string              `json:"vendorID,omitempty"`
-	Failed   bool                `json:"failed,omitempty"`
-	Status   string              `json:"status,omitempty"`
-	Config   *apiextensions.JSON `json:"config,omitempty"`
+	// DeviceID is the unique identifier for the hardware device.
+	// +optional
+	DeviceID string `json:"deviceID,omitempty"`
```

Comment:
I think DeviceID should be required @aneeshkp ?

Reply:
All new field should be optional , unless we  updated API . 
and introduce conversion webhooks are needed to convert between versions when necessary.